cardano-launcher/test/LauncherSpec.hs

```diff
@@ -139,4 +132,69 @@ setWorkingDirectorySpec = describe "Set working directory" $ do
+    describe "getLauncherOptions" $ do
+        mapM_ testGetLauncherOption launcherDatas
+
+        it "should fail due to passing wrong file path" $ monadicIO $ do
+            eLauncherOption <- run $ do
+                -- Path to the launcher file is incorrect
+                let launcherOptionPath = LauncherOptionPath "this will fail"
+                withSystemTempDirectory "test-XXXXXX" $ \tmpDir ->
+                    withSetEnvs launcherOptionPath tmpDir $ decodeLauncherOption launcherOptionPath
+            assert $ isLeft eLauncherOption
+
+        it "should fail due to missing env vars" $ monadicIO $ do
+            eLauncherOption <- run $ do
+                let launcherOptionPath = LauncherOptionPath (configFullFilePath "launcher-config-mainnet.linux.yaml")
+                -- Not environment variables are set!
+                decodeLauncherOption launcherOptionPath
+            assert $ isLeft eLauncherOption
+
+        it "should fail due to failing to convert to LauncherOption" $ monadicIO $ do
+            eLauncherOption <- run $ do
+                withSystemTempDirectory "test-XXXXXX" $ \tmpDir -> do
+                    -- Create invalid yaml file
+                    let val = String "this is not launcher option"
+                    let yamlPath = tmpDir </> "launcher.yaml"
+                    encodeFile yamlPath val
+                    let launcherOptionPath = LauncherOptionPath yamlPath
+                    withSetEnvs launcherOptionPath tmpDir $ decodeLauncherOption launcherOptionPath
+            assert $ isLeft eLauncherOption
+
+-- | Test that env var substitution works as expected on actual config files
+testGetLauncherOption :: FilePath -> Spec
+testGetLauncherOption configPath = it ("should be able to perform env substitution on config: " <> configPath) $ monadicIO $ do
+    eLauncherOption <- run $ do
+        let launcherOptionPath = LauncherOptionPath (configFullFilePath configPath)
+        withSystemTempDirectory "test-XXXXXX" $ \tmpDir ->
+            withSetEnvs launcherOptionPath tmpDir $ decodeLauncherOption launcherOptionPath
+    assert $ isRight eLauncherOption
+
+configFullFilePath :: FilePath -> FilePath
+configFullFilePath configPath = "./configuration/launcher/" <> configPath
+
+-- | Set all the envrionment variables that are needed to perform env var
+-- substituion then unset them afterwards
+withSetEnvs :: LauncherOptionPath -> FilePath -> IO a -> IO a
+withSetEnvs path homePath action = bracket
+    (do
+        -- Scripts that runs the launcher sets DAEDALUS_CONFIG variable
+        setEnv "DAEDALUS_CONFIG" "Foo"
+        -- Every operating system have HOME variable set
+        setEnv "HOME" homePath
+        setupEnvVars path
+    )
+    (const $ mapM_ unsetEnv
+        [ "DAEDALUS_CONFIG"
+        , "HOME"
+        , "XDG_DATA_HOME"
+        , "DAEDALUS_INSTALL_DIRECTORY"
+        , "LAUNCHER_CONFIG"
+        ]
+    )
+    (const $ action)
```

Comment:
This is fine, but I sometimes prefer to use expanded anonymous function definition instead of `const`, since it might be more readable - `\_ -> action`.
Any time I have to add an imaginary variable in my head make the code a bit more difficult.
Nothing wrong with it here, just my take on it.